server/chat/backend/agent/providers/bedrock_provider.py

```
@@ -137,7 +142,22 @@ def get_chat_model(
         kwargs.pop("streaming", None)
 
+        import time as _br_time
+
+        # Check client cache to avoid repeated boto3 session + TLS handshake
+        cache_key = f"{native_model}:{self.region}:{temperature}"
+        now = _br_time.time()
+        cached = BedrockProvider._native_client_cache.get(cache_key)
+        if cached:
+            client, created_at = cached
+            if now - created_at < BedrockProvider._CACHE_TTL:
+                logger.info(f"[LATENCY] Bedrock client cache HIT for {native_model} (region={self.region})")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)